src/compiler/parser.ts

```diff
@@ -4343,10 +4343,10 @@ namespace ts {
             }
 
             const pos = getNodePos();
-            let expr = parseAssignmentExpressionOrHigher();
+            let expr = parseAssignmentExpressionOrHigher(/*disallowReturnTypeInArrowFunction*/ false);

```

Comment:
Note: the `disallow` is throwingme off. Consider `allow`, especially since technically TypeScript scenarios (where we usually allow) are the exception.

Reply:
Yeah, I need to rename this to something more reasonable. I think this is like the third name choice and it's still not good enough.

Reply:
Maybe `canParseReturnTypeInArrowFunction`?

Reply:
`allowXxxYyyZzz` seems to be the more common naming scheme, so I'll just do that.